src/main/java/ua/okwine/productexpirationdate/controller/ProviderController.java

```diff
@@ -0,0 +1,96 @@
+package ua.okwine.productexpirationdate.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import ua.okwine.productexpirationdate.entity.Provider;
+import ua.okwine.productexpirationdate.service.ProviderService;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+
+@Controller
+@RequestMapping("/providers")
+public class ProviderController {
```

Comment:
Perhaps supplier would be better but it's up to you, it's related to the business logic and it's important that it's fitting you rather that even precise translation